src/test/java/unit/AiNotebookControlled.java

```diff
@@ -0,0 +1,312 @@
+
+    // ai notebooks are always private
+    assertEquals(
+        AccessScope.PRIVATE_ACCESS, createdNotebook.accessScope, "create output matches access");
+    assertEquals(
+        workspaceCreator.email.toLowerCase(),
+        createdNotebook.privateUserName.toLowerCase(),
+        "create output matches private user name");
+    // TODO (PF-616): check the private user roles once WSM returns them
+
+    // `terra resources describe --name=$name --format=json`
+    UFAiNotebook describeResource =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFAiNotebook.class, "resources", "describe", "--name=" + name);
+
+    // check that the properties match
+    assertEquals(name, describeResource.name, "describe resource output matches name");
+    assertEquals(cloning, describeResource.cloningInstructions, "describe output matches cloning");
+    assertEquals(description, describeResource.description, "describe output matches description");
+    assertEquals(location, describeResource.location, "describe resource output matches location");
+    assertEquals(
+        instanceId, describeResource.instanceId, "describe resource output matches instance id");
+
+    // ai notebooks are always private
+    assertEquals(
+        AccessScope.PRIVATE_ACCESS, describeResource.accessScope, "describe output matches access");
+    assertEquals(
+        workspaceCreator.email.toLowerCase(),
+        describeResource.privateUserName.toLowerCase(),
+        "describe output matches private user name");
+    // TODO (PF-616): check the private user roles once WSM returns them
+  }
+
+  @Test // NOTE: This test takes ~10 minutes to run.
+  @DisplayName("start, stop a notebook and poll until they complete")
+  void startStop() throws IOException, InterruptedException {
+    workspaceCreator.login();
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getWorkspaceId());
+
+    // `terra resources create ai-notebook --name=$name`
+    String name = "startStop";
+    TestCommand.runCommandExpectSuccess("resources", "create", "ai-notebook", "--name=" + name);
+    pollDescribeForNotebookState(name, "PROVISIONING");
+
+    // `terra notebooks start --name=$name`
+    // TODO (PF-869): change this to expect success once polling notebook operations is fixed
+    TestCommand.runCommand("notebooks", "start", "--name=" + name);
+    pollDescribeForNotebookState(name, "ACTIVE");
+
+    // `terra notebooks stop --name=$name`
+    // TODO (PF-869): change this to expect success once polling notebook operations is fixed
+    TestCommand.runCommand("notebooks", "stop", "--name=" + name);
+    pollDescribeForNotebookState(name, "STOPPED");
+  }
+
+  /**
+   * Helper method to poll `terra resources describe` until the notebook state equals that
+   * specified. Uses the current workspace.
+   */
+  public static void pollDescribeForNotebookState(String resourceName, String notebookState)
```

Comment:
If we're going to reuse these in multiple tests, make a shared notebook test utilities file.

Reply:
These utility methods are notebooks-related. I think it makes sense to put them in the `AiNotebookControlled` class, where they are used most. If we end up using them in many other classes, then I'd be up for moving them into a `utils` class, but since they're only used in one other place now, I think it's unnecessary.